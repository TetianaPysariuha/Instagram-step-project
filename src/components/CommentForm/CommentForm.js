/* eslint-disable no-underscore-dangle */
import React from 'react';
import {
  Formik, Form, Field,
} from 'formik';
import { useDispatch, useSelector } from 'react-redux';
import * as yup from 'yup';
import styles from './CommentForm.module.scss';
import { updatePost } from '../../store/posts/actionCreators';

function CommentForm({ postId }) {
  const dispatch = useDispatch();
  const posts = useSelector((store) => store.posts.posts);
  const loggedUser = useSelector((store) => store.users.loggedUser);

  const initialValues = {
    comment: '',
  };

  const validationChema = yup.object().shape({
    comment: yup.string().required('For adding a comment you need to write something'),
  });

  const handleSubmit = (values, { resetForm }) => {
    const newOne = posts.find((el) => el._id === postId);
    const newComments = newOne.comments;
    newComments.push({ userId: loggedUser._id, text: values.comment });
    dispatch(updatePost({ id: postId, data: { comments: newComments } }));
    resetForm();
  };

  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationChema}
      onSubmit={handleSubmit}
    >
      {({ isValid, dirty }) => (
        <Form className={styles.form}>

          <Field
            className={styles.formField}
            name="comment"
            type="text"
            placeholder="Напишіть коментар"
          />

          <button
            className={styles.btnSubmit}
            disabled={!dirty || !isValid}
            type="submit"
          >
            Опублікувати
          </button>
        </Form>
      )}
    </Formik>
  );
}

export default CommentForm;
