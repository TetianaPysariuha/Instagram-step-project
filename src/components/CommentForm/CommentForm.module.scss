*{
    padding: 0;
    margin: 0;
}
.form{
    width: 100%;
    height: 50px;
    display: grid;
    box-sizing: border-box;
    grid-template-columns: 1fr 100px;
}
.formField{
    padding-left: 10px;
    box-sizing: border-box;
}
.formField Field{
    border: 1px solid rgba($color: gray, $alpha: 0.3);
    height: 100%;
}
.btnSubmit{
    border: none;
    background-color: white;
    height: 100%;
    cursor: pointer;
    align-self: center;
    border: 1px solid rgba($color: gray, $alpha: 1);
}