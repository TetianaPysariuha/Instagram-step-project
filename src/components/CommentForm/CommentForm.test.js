import { render } from '@testing-library/react';
import CommentForm from './CommentForm';
import '@testing-library/jest-dom';

jest.mock('react-redux');

describe('CommentForm snapshot testing', () => {
  test('Should Modal to render', () => {
    const { asFragment } = render(<CommentForm postId="1" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
