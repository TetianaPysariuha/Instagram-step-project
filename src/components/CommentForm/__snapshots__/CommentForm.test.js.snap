// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm snapshot testing Should Modal to render 1`] = `
<DocumentFragment>
  <form
    action="#"
    class="form"
  >
    <input
      class="formField"
      name="comment"
      placeholder="write a comment"
      type="text"
      value=""
    />
    <button
      class="btnSubmit"
      disabled=""
      type="submit"
    >
      Add comment
    </button>
  </form>
</DocumentFragment>
`;
